File: examples/profile/tests/js/exp.js
// exponent test

function expLoop(n) {
	var sum = 0;
	for ( var i=0; i!=n; ++i ) {
		sum += Math.exp(1.0/(1.0+i));
	}
	return sum;
}

// infrastructure

function timeStamp() {
    if (typeof Duktape == 'object')
        return Date.now();
	var timeStampInMs =
		window.performance &&
		window.performance.now &&
		window.performance.timing &&
		window.performance.timing.navigationStart
			? window.performance.now() + window.performance.timing.navigationStart : Date.now();
	return timeStampInMs;
}

function profile(tname,cnt,testFn) {
	var t = 100500;
	var count = cnt
	while ( count>0 ) {
		var t0 = timeStamp();
		testFn();
		var t1 = timeStamp();
		var dt = t1 - t0;
		t = Math.min(dt, t);
		count --;
	}
	t /= 1000.0;
	var msg = '"'+tname+'", '+t+', '+cnt;
    if (typeof Duktape == 'object')
    {
        print(msg)
	} else
	{
	    msg+='\n'
        console.log(msg);
      	document.body.innerHTML += msg + "<br>";
  	}
}

function performance_tests() {
	profile("exp loop",20,function(){
		expLoop(1000000);
	});
	timeStamp();
}

performance_tests();

File: examples/profile/tests/cs/exp.cs
using System;
using System.Diagnostics;
using System.Timers;
using System.Collections.Generic;

class HelloWorld {

    delegate void MyBlock();

    static void profile ( int count, string category, MyBlock f ) {
        double minT = 1e+06;
        for ( int i = 0; i < count; i++ ) {
            DateTime start = DateTime.Now;
            f();
            DateTime end = DateTime.Now;
            TimeSpan elapsed = end - start;
            double dt = ((double)elapsed.TotalMilliseconds) / 1000.0;
            minT = Math.Min(minT, dt);
        }
        Console.WriteLine($"\"{category}\", {minT}, {count}");
    }

    // Translated expLoop function
    public static double ExpLoop(int n)
    {
        double sum = 0.0;
        for (int i = 1; i <= n; i++)
        {
            sum += Math.Exp(1.0 / (1.0 + i));
        }
        return sum;
    }

    // Translated verifyExp function
    public static void VerifyExp(double f)
    {
        double t = 1e+06;
        double eps = 10.0;
        double q = Math.Abs(f - t);
        Debug.Assert(q < eps, "The result is outside the acceptable tolerance.");
    }

    static void Main() {
        profile(20, "exp loop", () => {
            double f4 = ExpLoop(1000000);
            VerifyExp(f4);
        });
    }
}
